plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,698 @@
+                            "are not from the same controller/system.")
+
+        cmds = ["create", ctrl_num, "logicaldrive", "1024", arcconf_raid_level]
+        for disk_channel_device in arcconf_disk_ids:
+            cmds.append(disk_channel_device['Channel'])
+            cmds.append(disk_channel_device['Device'])
+
+        try:
+            self._arcconf_exec(cmds, flag_force=True)
+        except ExecError:
+            # Check whether disk is free
+            requested_disk_ids = [d.id for d in disks]
+            for cur_disk in self.disks():
+                if cur_disk.id in requested_disk_ids and \
+                   not cur_disk.status & Disk.STATUS_FREE:
+                    raise LsmError(
+                        ErrorNumber.DISK_NOT_FREE,
+                        "Disk %s is not in STATUS_FREE state" % cur_disk.id)
+            raise
+
+        # Generate pool_id from system id and array.
+        decoded_json = self._get_detail_info_list()[int(ctrl_num) - 1]
+
+        sys_id = ctrl_num
+        cnt = ctrl_num
+        try:
+            latest_ld = len(decoded_json['Controller']['LogicalDrive']) - 1
+            ld_num = \
+                (decoded_json['Controller']
+                    ['LogicalDrive']
+                    [latest_ld]
+                    ['logicalDriveID'])
+            ld_name = \
+                decoded_json['Controller']['LogicalDrive'][latest_ld]['name']
+            pool_id = '%s:%s' % (ctrl_num, ld_num)
+        except DeviceNotFoundError:
+            pass
+
+        lsm_vols = self.volumes(search_key='pool_id', search_value=pool_id)
+
+        if len(lsm_vols) < 1:
+            raise LsmError(
+                ErrorNumber.PLUGIN_BUG,
+                "volume_raid_create(): Got unexpected count(not 1) of new "
+                "volumes: %s" % lsm_vols)
+        return lsm_vols[0]
+
+    @_handle_errors
+    def volume_delete(self, volume, flags=0):
+        """
+        Depends on command:
+            arcconf delete <ctrlNo> logicaldrive <ld#> noprompt
+        """
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        (ctrl_num, array_num, ld_num) = volume.plugin_data.split(":")
+
+        try:
+            self._arcconf_exec(
+                ['delete',
+                    ctrl_num,
+                    'logicaldrive',
+                    array_num],
+                flag_force=True)
+        except ExecError:
+            ctrl_info = self._get_detail_info_list()[int(ctrl_num) - 1]
+            for ld in range(len(ctrl_info['LogicalDrive'])):
+                ld_info = ctrl_info['LogicalDrive']
+                if array_num == ld_info[ld]['logicalDriveID']:
+                    raise LsmError(
+                          ErrorNumber.PLUGIN_BUG,
+                          "volume_delete failed unexpectedly")
```

Comment:
If possible, provide more info besides `volume_delete failed unexpectedly`. It could be helpful when user report a bug to us.